server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/MySQLAffectedRowsExceedLimit.java

```diff
@@ -76,8 +74,7 @@ public long getStatementAffectedRows(Statement statement) {
             String explainSql = "EXPLAIN " + statement.getText();
             try {
                 if (this.jdbcOperations == null) {
-                    log.warn("JdbcOperations is null, please check your connection");
-                    return -1;
+                    throw new IllegalStateException("JdbcOperations is null, please check your connection");
```

Comment:
which caller will catch this exception?

Reply:
![image](https://github.com/user-attachments/assets/fe0188c7-d7b2-47c6-b565-dad151f24f15)
the input parameter jdbc is null when the SqlAffectedRowsFactory is constructed

Reply:
who will catch this exception? what will be happened at the front end if this exception is thrown?

Reply:
![image](https://github.com/user-attachments/assets/ce8f1b5a-e647-4161-bb79-ff9371850bc7)
The parent class's template method will catch the exception and engulf it

Reply:
so, why should we throw this exception rather that return -1?